static/app/utils/profiling/profile/sampledProfile.tsx

```diff
@@ -0,0 +1,119 @@
+import {lastOfArray} from 'sentry/utils';
+import {CallTreeNode} from 'sentry/utils/profiling/callTreeNode';
+
+import {Frame} from './../frame';
+import {Profile} from './profile';
+import {createFrameIndex} from './utils';
+
+export class SampledProfile extends Profile {
+  static FromProfile(sampled: Profiling.SampledProfile): Profile {
+    const {startValue, endValue, samples, weights} = sampled;
+
+    const sampledProfile = new SampledProfile(
+      endValue - startValue,
+      startValue,
+      endValue,
+      sampled.name,
+      sampled.unit
+    );
+
+    const frameIndex = createFrameIndex(sampled.shared.frames);
+
+    if (samples.length !== weights.length) {
+      throw new Error(
+        `Expected samples.length (${samples.length}) to equal weights.length (${weights.length})`
+      );
+    }
+
+    for (let i = 0; i < samples.length; i++) {
+      const stack = samples[i];
+      const weight = weights[i];
+
+      sampledProfile.appendSampleWithWeight(
+        stack.map(n => frameIndex[n]),
+        weight
+      );
+    }
+
+    return sampledProfile.build();
+  }
+
+  appendSampleWithWeight(stack: Frame[], weight: number): void {
+    // Ignore samples with 0 weight
+    if (weight === 0) return;
+    if (isNaN(weight)) throw new Error('invalid weight');
```

Comment:
In what scenarios can the weight be `NaN`?

Reply:
@Zylphrex probably can’t happen, I think this was some old guard.